content/en/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -36,10 +36,12 @@ No parameters are passed to the handler.
 
 `PreStop`
 
-This hook is called immediately before a container is terminated due to an API request or management event such as liveness probe failure, preemption, resource contention and others. A call to the preStop hook fails if the container is already in terminated or completed state.
-It is blocking, meaning it is synchronous,
-so it must complete before the signal to stop the container can be sent.
-No parameters are passed to the handler.
+This hook is called immediately before a container is terminated due to an API request or management
+event such as a liveness/startup probe failure, preemption, resource contention and others. This
+hook is not called if the container crashes or exits. The Pod's termination grace period countdown
+begins before the `PreStop` hook is executed. Regardless of the outcome of the handler, the
+container will eventually terminate within the Pod's termination grace period. No parameters are
+passed to the handler.
```

Comment:
Two comments/questions on this section:

1) I would like to include the phrase "The hook must complete before the signal to stop the container can be sent" per the original.

2) Where are you getting the information that `Regardless of the outcome of the handler, the
container will eventually terminate within the Pod's termination grace period.` happens? 

3) I'd like to rephrase `This hook is not called if the container crashes or exits` back to the original (ish): "A call to the preStop hook fails if the container is already in a terminated or completed state." – the wording "crashes or exits" doesn't reflect actual states a container can be in and is therefore confusing.

Reply:
1. Done
2. From the output of `kubectl explain pod.spec.containers.lifecycle.preStop`
3. Done